# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSContext.cpp
# Commit: 5c0565636d06
# Full Hash: 5c0565636d06c53ae0d4455158164da0f3d9cb6a
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-13 21:46:27
# Regressor Bug: 1634135
# File Overlap Count: 2
# Description:
#   Bug 1634135: Don't leak Isolate r=mgaudet
#   
#   ASAN found a leak.
#   
#   We destroy the isolate with a level of indirection through RegexpAPI so that JSContext doesn't have to be able to see the full definition of Isolate.
# ==============================================================================

diff -r 16a1d1bbdd3e -r 5c0565636d06 js/src/vm/JSContext.cpp
--- a/js/src/vm/JSContext.cpp	Wed May 13 16:22:55 2020 +0000
+++ b/js/src/vm/JSContext.cpp	Wed May 13 16:23:20 2020 +0000
@@ -974,6 +974,10 @@
   }
 #endif
 
+#ifdef ENABLE_NEW_REGEXP
+  irregexp::DestroyIsolate(isolate.ref());
+#endif
+
   js_delete(atomsZoneFreeLists_.ref());
 
   TlsContext.set(nullptr);
